app/browser/reducers/passwordManagerReducer.js

```diff
@@ -189,14 +189,17 @@ const migrate = (state) => {
                                                password.get('authTag'),
                                                masterKey,
                                                password.get('iv'))
-      let form = {}
-      form['origin'] = password.get('origin')
-      form['signon_realm'] = password.get('origin') + '/'
-      form['action'] = password.get('action')
-      form['username'] = password.get('username')
-      form['password'] = decrypted
-      autofill.addLogin(form)
+      if (decrypted) {
+        let form = {}
+        form['origin'] = password.get('origin')
+        form['signon_realm'] = password.get('origin') + '/'
+        form['action'] = password.get('action')
+        form['username'] = password.get('username')
+        form['password'] = decrypted
+        autofill.addLogin(form)
+      }
     })
+    state = state.set('legacy_passwords', state.get('passwords'))
```

Comment:
Will do. 
These two state set lines are under `if (passwords.size) {...}` which use to check non-migrated data and I think inside of `passwords.forEach((password) => {...})` is called once per password